packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -355,6 +361,55 @@ def __eq__(self, other: Any) -> bool:
         return self.payload == other.payload and self.signature == other.signature
 
 
+@dataclass(frozen=True)
+class NewBaseTxPayload(ABC, metaclass=_MetaPayload):
+    """This class represents a base class for transaction payload classes."""
+
+    sender: str
+    round_count: int = field(default=ROUND_COUNT_DEFAULT, init=False)
```

Comment:
just a plain dataclass and nothing more. No extra methods for serialization or anything. 
- The `round_count` field cannot be set during initialization and hence will have the default value, which we set in `send_a2a_transaction`. Reason for providing it like this has two reasons:
  1. `round_count` is for internal use and not generally relevant to devs, neither should they set it in the first place.
      https://github.com/valory-xyz/open-autonomy/blob/4d623d97896c7ad3675cb82de07d3dcd7099c38a/packages/valory/skills/abstract_round_abci/behaviour_utils.py#L703
  2. providing a default with `init=False` allows to subclass without the necessity to provide default value (would be necessary without default value) and without having to specify the `round_count`. To illustrate
        ```
        @dataclass(frozen=True)
        class MyPayload(NewBaseTxPayload):
            a : int
        ```
        would not be possible to initialize as follows
        ```
        MyPayload("s", 2)
        >>> TxPay(sender='s', round_count=-1, a=2)
        ```
- 


 Current implementation here
https://github.com/valory-xyz/open-autonomy/blob/4d623d97896c7ad3675cb82de07d3dcd7099c38a/packages/valory/skills/abstract_round_abci/base.py#L205-L311